### PR TITLE
Click on the phone number and Serial Number to add to clipboard in settings app

### PR DESCRIPTION
**Pull request by mrdiamonddirt**
## Description
Phone Number and Serial Number copied to clipboard when clicked in settings page
image

## Checklist

- [x] I have personally loaded this code into an updated qbcore project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Comment by gononono64** [comment]
Looks solid. This should be added because often, people want to send or use their in-game phone number outside of fivem.  Ill test it in a few

**Review by Z3rio — CHANGES_REQUESTED** [review]
I dont really think we should be importing yet another library just for something which can be done in 2-3 lines of standard JS, aka setting the clipboard.

QB-Phone is already bloated enough as it is...

**Comment by gononono64** [comment]
> I dont really think we should be importing yet another library just for something which can be done in 2-3 lines of standard JS, aka setting the clipboard.
> 
> QB-Phone is already bloated enough as it is...

Imo copy to clipboard should be part of core

**Comment by Z3rio** [comment]
> > I dont really think we should be importing yet another library just for something which can be done in 2-3 lines of standard JS, aka setting the clipboard.
> > QB-Phone is already bloated enough as it is...
> 
> Imo copy to clipboard should be part of core

Apart of qb-core? Yeah, perhaps, would be nice when trying to set the clipboard from lua.

Wouldnt really be useable/ideal in this situation even if it was in qb-core though.
As you'd have to go from the UI, to the clientside of qb-phone, then to the clientside of qb-core, then to the UI of qb-core.

Much better to just write those 2-3 lines of JS in qb-phone in this situation.

**Comment by mrdiamonddirt** [comment]
Absolutely 👍 didn't spend too much time with it but I would've thought i'd tried it without a library and thought it didn't work for some reason so probably did that naturally I thought I could do it with native js but not looked at this really since posting was just a quick implementation that worked

**Comment by mrdiamonddirt** [comment]
I might pull it down and have a look again and see why I didn't use the clipboard I'm sure I remember trying it, but someone could always pull it down remove the package add the 2-3 lines of js and that I maybe missed and create the pr again and comment on this to say it's outdated because of pr # I don't have or use qb-phone and am not a "qb-core developer" obviously this isn't top of list of things for me todo 😀 but I appreciate the comments and the response @Z3rio and any constructive criticism is always appreciated

**Comment by Z3rio** [comment]
> I might pull it down and have a look again and see why I didn't use the clipboard I'm sure I remember trying it, but someone could always pull it down remove the package add the 2-3 lines of js and that I maybe missed and create the pr again and comment on this to say it's outdated because of pr # I don't have or use qb-phone and am not a "qb-core developer" obviously this isn't top of list of things for me todo 😀 but I appreciate the comments and the response @Z3rio and any constructive criticism is always appreciated

Yeah, well, if qb-core actually cared about PRs and merged them, I'd 110% do that.
But I already have 6 PRs of my own waiting on being merged lmfao

**Comment by mrdiamonddirt** [comment]
I completely get that buddy I also like to not have a load of outstanding pr's, so I keep them few and fairly minimal my argument for this way could be yeah it adds 25 lines and a libary but it is a very minimal libary and makes re-use of the copy function I implemented to be very easily re-used elsewhere, yes it may able to be done more minimally, but I honestly like this implementation,😂🤣 and I'm not flaming your comments at all @Z3rio they are obviously valid

**Review by ChristianBDev — CHANGES_REQUESTED** [review]
Can you add this to the settings.js, line 38 
```
$(document).on(
    "click",
    "#phoneNumberSelect, #serialNumberSelect",
    function (e) {
        // Get the title of the clicked element
        var title = "";
        if ($(this).attr("id") == "phoneNumberSelect") {
            title = "Phone Number";
        } else {
            title = "Serial Number";
        }

        // get the result id of myPhoneNumber or mySerialNumber
        var textToCopy =
            $(this).attr("id") == "phoneNumberSelect"
                ? $("#myPhoneNumber").text()
                : $("#mySerialNumber").text();

        // Copying the text to clipboard using Clipboard.js
        var clipboard = new ClipboardJS(this, {
            text: function () {
                QB.Phone.Notifications.Add("fas fa-phone", "Copied " + title + "!", textToCopy);
                return textToCopy;
            },
        });
    }
);
``` 
and delete the line on 1135 
`<script src="./js/copyclipboard.js"></script>`

File: html/index.html
Comment (ChristianBDev) [inline html/index.html:1135]: Remove this

File: html/js/copyclipboard.js
Comment (ChristianBDev) [inline html/js/copyclipboard.js:20]: remove this

**Review by ChristianBDev — CHANGES_REQUESTED** [review]
Add this to the settings.js instead ```
$(document).on(
    "click",
    "#phoneNumberSelect, #serialNumberSelect",
    function (e) {
        // Get the title of the clicked element
        var title = "";
        if ($(this).attr("id") == "phoneNumberSelect") {
            title = "Phone Number";
        } else {
            title = "Serial Number";
        }

        // get the result id of myPhoneNumber or mySerialNumber
        var textToCopy =
            $(this).attr("id") == "phoneNumberSelect"
                ? $("#myPhoneNumber").text()
                : $("#mySerialNumber").text();

        // Copying the text to clipboard using Clipboard.js
        var clipboard = new ClipboardJS(this, {
            text: function () {
                QB.Phone.Notifications.Add("fas fa-phone", "Copied " + title + "!", textToCopy);
                return textToCopy;
            },
        });
    }
);```

**Review by ChristianBDev — APPROVED** [review]
Tested, LGTM

**Pull request merged**
